cylc/flow/platforms.py

```diff
@@ -24,6 +24,94 @@
 from cylc.flow.cfgspec.glbl_cfg import glbl_cfg
 
 
+def get_platform(task_conf=None):
+    """A single entry point for this module: Looking at a task config this
+    method decides whether to get platform from name, or Cylc7 config items.
+
+    Args:
+        task_conf (str, dict or dict-like such as OrderedDictWithDefaults):
+            If str this is assumed to be the platform name, otherwise this
+            should be a configuration for a task.
+
+    Returns:
+        platform:
+            Actually it returns either get_platform() or
+            platform_from_job_info(), but to the user these look the same.
+
+    """
+
+    forbidden_with_platform = (
+        ('remote', 'host', ['localhost', None]),
+        ('job', 'batch system', [None]),
+        ('job', 'batch submit command template', [None])
+    )
+
+    if task_conf is None:
+        # Just a simple way of accessing localhost items.
+        output = platform_from_name()
+
+    elif type(task_conf) is str:
+        # If task_conf is str assume that it is a platform name.
+        output = platform_from_name(task_conf)
+
+    elif 'platform' in task_conf and task_conf['platform']:
+        # If platform is set check whether any Cylc7 items are present -
+        # and fail if they are:
+        fail_items = ''
+        for section, key, _ in forbidden_with_platform:
+            if (
+                section in task_conf and
+                key in task_conf[section] and
+                task_conf[section][key]
+            ):
+                fail_items += (
+                    f'- task_conf[section][key] is '
+                    f' [{section}]{key} = {task_conf[section][key]}\n'
+                )
+            if fail_items:
+                raise PlatformLookupError(
+                    "A mixture of Cylc 7 (host) and Cylc 8 (platform)"
+                    " logic should not be used. In this case the "
+                    "following are not compatible\n" +
+                    fail_items
+                )
```

Comment:
```suggestion
                raise PlatformLookupError(
                    f"A mixture of Cylc 7 (host) and Cylc 8 (platform) "
                    f"logic should not be used. In this case the "
                    f"following are not compatible\n{fail_items}"
                )
```
Use f-strings instead of string concatenation